OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -606,7 +607,18 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 							return true;
 						}
 
-						targetClient.Faction = parts[1];
+						var newFaction = parts[1];
+						targetClient.Faction = newFaction;
+
+						if (targetClient.Index == client.Index && newFaction == client.Faction)
+						{
+							client.PreferredFaction = client.Faction;
+
+							// Save preferred faction to settings.
+							Game.Settings.Game.LastFaction[Game.ModData.Manifest.Id] = client.PreferredFaction;
```

Comment:
All the code in this file runs on the server, so you can't modify per-user settings here.